internal/agent/upgrade.go

```diff
@@ -238,12 +238,14 @@ func upgradeUki(source string, dirs []string, upgradeEntry string, strictValidat
 // ExtraConfigUpgrade is the struct that holds the upgrade options that come from flags and events
 type ExtraConfigUpgrade struct {
```

Comment:
We take the cli args and we make a new type of config (this one). Then we serialize this thing to a json (or YAML?) string and we merge it with the rest of the config. The keys of the resulting string didn't match those of the Config we are trying to build thus the setting only made it to the raw Config (which we store as a separate key). That's why the original PR (see the first commit in this one) had to use `cfg.Query` to get to this key.

With the changes here, the actual fields of the Config get populated correctly.